### DISCUSSION TITLE
Is there a need of polyfills now for es5 Array methods

### DISCUSSION THREAD

**Discussion by AnilSeervi** [mdn/mdn-community#604] [Content]
#### What information was incorrect, unhelpful, or incomplete?
I think the polyfill isn't necessary now as es5 supported by most of the browsers now.
If it is to be removed then I'd like to work on it.

Some other pages that have the polyfills :
- [x] `Array.some()`
- [x] `Array.map()`
- [x] `Array.lastIndexOf()`
- [x] `Array.every()`
- [x] `Array.reduceRight()`

**Comment by wbamberg** [comment]
> #### What information was incorrect, unhelpful, or incomplete?
> 
> I think the polyfill isn't necessary now as es5 supported by most of the browsers now. If it is to be removed then I'd like to work on it.
> 
> Some other pages that have the polyfills :
> 
> * `Array.some()`
> 
> * `Array.map()`
> 
> * `Array.lastIndexOf()`
> 
> * `Array.every()`
> 
> * `Array.reduceRight()`

It's a good question. As you might know we have a proposed project to sort out polyfills on MDN, that didn't get prioritised in this quarter but which I do expect we will get to in 2022 (https://github.com/openwebdocs/project/issues/27). As part of that @teoli2003 started a discussion about when we should have any polyfills at all: https://github.com/mdn/content/discussions/7841 and by his proposed criteria we should remove them at least for the 5 pages you've listed here, which are all supported from IE9.
 
So I'd say: yes, we can remove these. I'll drop a link to this issue in the discussion.

**Comment by zloirock** [comment]
Funny. Do you really think that those features are absolutely the same as in ES5 and polyfills for them are not required?

- `@@species` pattern, proper subclassing support, `ToLength` conversion instead of `ToUint32` on `.length`, other cases of modern semantic were added to those methods in modern browsers only in 2017.
- Polyfills also are bugfixes. For example, `.reduce` / `.reduceRight` [were broken in Chrome 80-82](https://bugs.chromium.org/p/chromium/issues/detail?id=1049982) - it's 2020.

**Comment by wbamberg** [comment]
> Funny. Do you really think...

I'd appreciate it if you could communicate politely in this repository.

**Comment by zloirock** [comment]
Sorry, but that really made me laugh.

**Comment by zloirock** [comment]
However, I agree that *naive* polyfills posted directly on MDN that do not cover modern cases and sometimes even are harmful should be removed.

**Comment by ddbeck** [comment]
@zloirock I'd like to remind you that this project is covered by the [Mozilla Community Participation Guidelines](https://www.mozilla.org/en-US/about/governance/policies/participation/). I want to call specific attention to the [_Expected Behavior_ section](https://www.mozilla.org/en-US/about/governance/policies/participation/#expected-behavior). Sarcastic comments aren't welcome here. There are lots of ways to make a point without being disrespectful to other contributors.

**Comment by zloirock** [comment]
In my comment, I do not see anything offensive or disrespectful. Instead of being offended by sarcastic comments, it is better to avoid the cause of their appearance.

**Comment by ddbeck** [comment]
@zloirock This is not about offense or about how you see it. This is about behavior that is disruptive, regardless of your intent or self-perception. I'd suggest that you move on from this issue. If you have questions about how to follow the community guidelines, then you can email me at dbeck@mozilla.com.

**Comment by pepelsbey** [comment]
Closing as outdated/irrelevant.
